integration/hypershift_test.go

```diff
@@ -0,0 +1,48 @@
+package integration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/testr"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+	hypershiftv1alpha1 "package-operator.run/package-operator/internal/controllers/hostedclusters/hypershift/v1alpha1"
+)
+
+func TestHyperShift(t *testing.T) {
+	// tests that PackageOperator will deploy a new remote-phase-manager
+	// for every ready HyperShift HostedCluster.
+	ctx := logr.NewContext(context.Background(), testr.New(t))
+
+	hc := &hypershiftv1alpha1.HostedCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-hc",
+			Namespace: "default",
+		},
+	}
+	err := Client.Create(ctx, hc)
+	require.NoError(t, err)
+	defer cleanupOnSuccess(ctx, t, hc)
+
```

Comment:
Is there a way to check here that no package was created?

Reply:
Negative assertions are not that nice,  because we can only set them up time-based.
E.g. in 10s no Package was created, as we are having this completely covered with unittests, just testing the success end-to-end test is ok.